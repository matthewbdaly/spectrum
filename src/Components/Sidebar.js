// @flow
import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import './Sidebar.scss';

type Props = {
  links: array,
  active: boolean
};

type State = {
  active: boolean
};

class Sidebar extends Component<Props, State> {
  constructor(props: Props) {
    super(props);
    this.state = {
      active: this.props.active ? true : false
    };
  }
  render() {
    let linkNodes = this.props.links.map((item, index) => {
      return (
        <li key={index}>
          <Link to={item.route}>{item.text}</Link>
        </li>
      );
    });
    return (
      <ul className={ 'sidebar' + (this.state.active ? ' active' : '') }>
        {linkNodes}
      </ul>
    );
  }
}

export default Sidebar;
