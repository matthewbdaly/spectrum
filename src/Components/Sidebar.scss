@import "../variables";

.sidebar {
    height: 100%;
    width: 0;
    position: fixed;
    z-index: 1;
    top: 0;
    left: 0;
    overflow-x: hidden;
    margin: 0;
    padding: 0;
    transition: 0.4s;
    background-color: $background;

    &.active {
        width: 250px;
    }

    li {
        list-style-type: none;
        width: 100%;

        a {
            border: 1px solid lightgray;
            padding: 20px;
            text-decoration: none;
            display: block;
        }
    }
}
