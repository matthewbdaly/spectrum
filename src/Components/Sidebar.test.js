import React from 'react';
import ReactDOM from 'react-dom';
import Sidebar from './Sidebar';
import { BrowserRouter as Router } from 'react-router-dom';

const sidebarlinks = [{
  route: "/alerts",
  text: "Alerts"
}, {
  route: "/bars",
  text: "Bars"
}, {
  route: "/buttons",
  text: "Buttons"
}, {
  route: "/cards",
  text: "Cards"
}, {
  route: "/list-views",
  text: "List Views"
}, {
  route: "/tab-bars",
  text: "Tab Bars"
}, {
  route: "/loader",
  text: "Loader"
}, {
  route: "/demo",
  text: "Demo"
}];

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(
    <Router>
      <Sidebar links={sidebarlinks} />
    </Router>,
    div
  );
});
