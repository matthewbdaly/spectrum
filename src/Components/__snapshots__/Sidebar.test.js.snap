// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the snapshot 1`] = `
Sidebar {
  "_reactInternalFiber": FiberNode {
    "_debugID": 23,
    "_debugIsCurrentlyTiming": false,
    "_debugOwner": null,
    "_debugSource": Object {
      "fileName": "/home/matthew/Projects/spectrum/src/Components/Sidebar.test.js",
      "lineNumber": 38,
    },
    "alternate": null,
    "child": FiberNode {
      "_debugID": 24,
      "_debugIsCurrentlyTiming": false,
      "_debugOwner": [Circular],
      "_debugSource": Object {
        "fileName": "/home/matthew/Projects/spectrum/src/Components/Sidebar.js",
        "lineNumber": 30,
      },
      "alternate": null,
      "child": FiberNode {
        "_debugID": 25,
        "_debugIsCurrentlyTiming": false,
        "_debugOwner": [Circular],
        "_debugSource": Object {
          "fileName": "/home/matthew/Projects/spectrum/src/Components/Sidebar.js",
          "lineNumber": 24,
        },
        "alternate": null,
        "child": FiberNode {
          "_debugID": 33,
          "_debugIsCurrentlyTiming": false,
          "_debugOwner": [Circular],
          "_debugSource": Object {
            "fileName": "/home/matthew/Projects/spectrum/src/Components/Sidebar.js",
            "lineNumber": 25,
          },
          "alternate": null,
          "child": null,
          "effectTag": 0,
          "expirationTime": 0,
          "firstEffect": null,
          "index": 0,
          "internalContextTag": 0,
          "key": null,
          "lastEffect": null,
          "memoizedProps": Object {
            "children": "Alerts",
            "href": "/alerts",
          },
          "memoizedState": null,
          "nextEffect": null,
          "pendingProps": null,
          "ref": null,
          "return": [Circular],
          "sibling": null,
          "stateNode": <a
            href="/alerts"
>
            Alerts
</a>,
          "tag": 5,
          "type": "a",
          "updateQueue": null,
        },
        "effectTag": 0,
        "expirationTime": 0,
        "firstEffect": null,
        "index": 0,
        "internalContextTag": 0,
        "key": "0",
        "lastEffect": null,
        "memoizedProps": Object {
          "children": <a
            href="/alerts"
>
            Alerts
</a>,
        },
        "memoizedState": null,
        "nextEffect": null,
        "pendingProps": null,
        "ref": null,
        "return": [Circular],
        "sibling": FiberNode {
          "_debugID": 26,
          "_debugIsCurrentlyTiming": false,
          "_debugOwner": [Circular],
          "_debugSource": Object {
            "fileName": "/home/matthew/Projects/spectrum/src/Components/Sidebar.js",
            "lineNumber": 24,
          },
          "alternate": null,
          "child": FiberNode {
            "_debugID": 34,
            "_debugIsCurrentlyTiming": false,
            "_debugOwner": [Circular],
            "_debugSource": Object {
              "fileName": "/home/matthew/Projects/spectrum/src/Components/Sidebar.js",
              "lineNumber": 25,
            },
            "alternate": null,
            "child": null,
            "effectTag": 0,
            "expirationTime": 0,
            "firstEffect": null,
            "index": 0,
            "internalContextTag": 0,
            "key": null,
            "lastEffect": null,
            "memoizedProps": Object {
              "children": "Bars",
              "href": "/bars",
            },
            "memoizedState": null,
            "nextEffect": null,
            "pendingProps": null,
            "ref": null,
            "return": [Circular],
            "sibling": null,
            "stateNode": <a
              href="/bars"
>
              Bars
</a>,
            "tag": 5,
            "type": "a",
            "updateQueue": null,
          },
          "effectTag": 0,
          "expirationTime": 0,
          "firstEffect": null,
          "index": 1,
          "internalContextTag": 0,
          "key": "1",
          "lastEffect": null,
          "memoizedProps": Object {
            "children": <a
              href="/bars"
>
              Bars
</a>,
          },
          "memoizedState": null,
          "nextEffect": null,
          "pendingProps": null,
          "ref": null,
          "return": [Circular],
          "sibling": FiberNode {
            "_debugID": 27,
            "_debugIsCurrentlyTiming": false,
            "_debugOwner": [Circular],
            "_debugSource": Object {
              "fileName": "/home/matthew/Projects/spectrum/src/Components/Sidebar.js",
              "lineNumber": 24,
            },
            "alternate": null,
            "child": FiberNode {
              "_debugID": 35,
              "_debugIsCurrentlyTiming": false,
              "_debugOwner": [Circular],
              "_debugSource": Object {
                "fileName": "/home/matthew/Projects/spectrum/src/Components/Sidebar.js",
                "lineNumber": 25,
              },
              "alternate": null,
              "child": null,
              "effectTag": 0,
              "expirationTime": 0,
              "firstEffect": null,
              "index": 0,
              "internalContextTag": 0,
              "key": null,
              "lastEffect": null,
              "memoizedProps": Object {
                "children": "Buttons",
                "href": "/buttons",
              },
              "memoizedState": null,
              "nextEffect": null,
              "pendingProps": null,
              "ref": null,
              "return": [Circular],
              "sibling": null,
              "stateNode": <a
                href="/buttons"
>
                Buttons
</a>,
              "tag": 5,
              "type": "a",
              "updateQueue": null,
            },
            "effectTag": 0,
            "expirationTime": 0,
            "firstEffect": null,
            "index": 2,
            "internalContextTag": 0,
            "key": "2",
            "lastEffect": null,
            "memoizedProps": Object {
              "children": <a
                href="/buttons"
>
                Buttons
</a>,
            },
            "memoizedState": null,
            "nextEffect": null,
            "pendingProps": null,
            "ref": null,
            "return": [Circular],
            "sibling": FiberNode {
              "_debugID": 28,
              "_debugIsCurrentlyTiming": false,
              "_debugOwner": [Circular],
              "_debugSource": Object {
                "fileName": "/home/matthew/Projects/spectrum/src/Components/Sidebar.js",
                "lineNumber": 24,
              },
              "alternate": null,
              "child": FiberNode {
                "_debugID": 36,
                "_debugIsCurrentlyTiming": false,
                "_debugOwner": [Circular],
                "_debugSource": Object {
                  "fileName": "/home/matthew/Projects/spectrum/src/Components/Sidebar.js",
                  "lineNumber": 25,
                },
                "alternate": null,
                "child": null,
                "effectTag": 0,
                "expirationTime": 0,
                "firstEffect": null,
                "index": 0,
                "internalContextTag": 0,
                "key": null,
                "lastEffect": null,
                "memoizedProps": Object {
                  "children": "Cards",
                  "href": "/cards",
                },
                "memoizedState": null,
                "nextEffect": null,
                "pendingProps": null,
                "ref": null,
                "return": [Circular],
                "sibling": null,
                "stateNode": <a
                  href="/cards"
>
                  Cards
</a>,
                "tag": 5,
                "type": "a",
                "updateQueue": null,
              },
              "effectTag": 0,
              "expirationTime": 0,
              "firstEffect": null,
              "index": 3,
              "internalContextTag": 0,
              "key": "3",
              "lastEffect": null,
              "memoizedProps": Object {
                "children": <a
                  href="/cards"
>
                  Cards
</a>,
              },
              "memoizedState": null,
              "nextEffect": null,
              "pendingProps": null,
              "ref": null,
              "return": [Circular],
              "sibling": FiberNode {
                "_debugID": 29,
                "_debugIsCurrentlyTiming": false,
                "_debugOwner": [Circular],
                "_debugSource": Object {
                  "fileName": "/home/matthew/Projects/spectrum/src/Components/Sidebar.js",
                  "lineNumber": 24,
                },
                "alternate": null,
                "child": FiberNode {
                  "_debugID": 37,
                  "_debugIsCurrentlyTiming": false,
                  "_debugOwner": [Circular],
                  "_debugSource": Object {
                    "fileName": "/home/matthew/Projects/spectrum/src/Components/Sidebar.js",
                    "lineNumber": 25,
                  },
                  "alternate": null,
                  "child": null,
                  "effectTag": 0,
                  "expirationTime": 0,
                  "firstEffect": null,
                  "index": 0,
                  "internalContextTag": 0,
                  "key": null,
                  "lastEffect": null,
                  "memoizedProps": Object {
                    "children": "List Views",
                    "href": "/list-views",
                  },
                  "memoizedState": null,
                  "nextEffect": null,
                  "pendingProps": null,
                  "ref": null,
                  "return": [Circular],
                  "sibling": null,
                  "stateNode": <a
                    href="/list-views"
>
                    List Views
</a>,
                  "tag": 5,
                  "type": "a",
                  "updateQueue": null,
                },
                "effectTag": 0,
                "expirationTime": 0,
                "firstEffect": null,
                "index": 4,
                "internalContextTag": 0,
                "key": "4",
                "lastEffect": null,
                "memoizedProps": Object {
                  "children": <a
                    href="/list-views"
>
                    List Views
</a>,
                },
                "memoizedState": null,
                "nextEffect": null,
                "pendingProps": null,
                "ref": null,
                "return": [Circular],
                "sibling": FiberNode {
                  "_debugID": 30,
                  "_debugIsCurrentlyTiming": false,
                  "_debugOwner": [Circular],
                  "_debugSource": Object {
                    "fileName": "/home/matthew/Projects/spectrum/src/Components/Sidebar.js",
                    "lineNumber": 24,
                  },
                  "alternate": null,
                  "child": FiberNode {
                    "_debugID": 38,
                    "_debugIsCurrentlyTiming": false,
                    "_debugOwner": [Circular],
                    "_debugSource": Object {
                      "fileName": "/home/matthew/Projects/spectrum/src/Components/Sidebar.js",
                      "lineNumber": 25,
                    },
                    "alternate": null,
                    "child": null,
                    "effectTag": 0,
                    "expirationTime": 0,
                    "firstEffect": null,
                    "index": 0,
                    "internalContextTag": 0,
                    "key": null,
                    "lastEffect": null,
                    "memoizedProps": Object {
                      "children": "Tab Bars",
                      "href": "/tab-bars",
                    },
                    "memoizedState": null,
                    "nextEffect": null,
                    "pendingProps": null,
                    "ref": null,
                    "return": [Circular],
                    "sibling": null,
                    "stateNode": <a
                      href="/tab-bars"
>
                      Tab Bars
</a>,
                    "tag": 5,
                    "type": "a",
                    "updateQueue": null,
                  },
                  "effectTag": 0,
                  "expirationTime": 0,
                  "firstEffect": null,
                  "index": 5,
                  "internalContextTag": 0,
                  "key": "5",
                  "lastEffect": null,
                  "memoizedProps": Object {
                    "children": <a
                      href="/tab-bars"
>
                      Tab Bars
</a>,
                  },
                  "memoizedState": null,
                  "nextEffect": null,
                  "pendingProps": null,
                  "ref": null,
                  "return": [Circular],
                  "sibling": FiberNode {
                    "_debugID": 31,
                    "_debugIsCurrentlyTiming": false,
                    "_debugOwner": [Circular],
                    "_debugSource": Object {
                      "fileName": "/home/matthew/Projects/spectrum/src/Components/Sidebar.js",
                      "lineNumber": 24,
                    },
                    "alternate": null,
                    "child": FiberNode {
                      "_debugID": 39,
                      "_debugIsCurrentlyTiming": false,
                      "_debugOwner": [Circular],
                      "_debugSource": Object {
                        "fileName": "/home/matthew/Projects/spectrum/src/Components/Sidebar.js",
                        "lineNumber": 25,
                      },
                      "alternate": null,
                      "child": null,
                      "effectTag": 0,
                      "expirationTime": 0,
                      "firstEffect": null,
                      "index": 0,
                      "internalContextTag": 0,
                      "key": null,
                      "lastEffect": null,
                      "memoizedProps": Object {
                        "children": "Loader",
                        "href": "/loader",
                      },
                      "memoizedState": null,
                      "nextEffect": null,
                      "pendingProps": null,
                      "ref": null,
                      "return": [Circular],
                      "sibling": null,
                      "stateNode": <a
                        href="/loader"
>
                        Loader
</a>,
                      "tag": 5,
                      "type": "a",
                      "updateQueue": null,
                    },
                    "effectTag": 0,
                    "expirationTime": 0,
                    "firstEffect": null,
                    "index": 6,
                    "internalContextTag": 0,
                    "key": "6",
                    "lastEffect": null,
                    "memoizedProps": Object {
                      "children": <a
                        href="/loader"
>
                        Loader
</a>,
                    },
                    "memoizedState": null,
                    "nextEffect": null,
                    "pendingProps": null,
                    "ref": null,
                    "return": [Circular],
                    "sibling": FiberNode {
                      "_debugID": 32,
                      "_debugIsCurrentlyTiming": false,
                      "_debugOwner": [Circular],
                      "_debugSource": Object {
                        "fileName": "/home/matthew/Projects/spectrum/src/Components/Sidebar.js",
                        "lineNumber": 24,
                      },
                      "alternate": null,
                      "child": FiberNode {
                        "_debugID": 40,
                        "_debugIsCurrentlyTiming": false,
                        "_debugOwner": [Circular],
                        "_debugSource": Object {
                          "fileName": "/home/matthew/Projects/spectrum/src/Components/Sidebar.js",
                          "lineNumber": 25,
                        },
                        "alternate": null,
                        "child": null,
                        "effectTag": 0,
                        "expirationTime": 0,
                        "firstEffect": null,
                        "index": 0,
                        "internalContextTag": 0,
                        "key": null,
                        "lastEffect": null,
                        "memoizedProps": Object {
                          "children": "Demo",
                          "href": "/demo",
                        },
                        "memoizedState": null,
                        "nextEffect": null,
                        "pendingProps": null,
                        "ref": null,
                        "return": [Circular],
                        "sibling": null,
                        "stateNode": <a
                          href="/demo"
>
                          Demo
</a>,
                        "tag": 5,
                        "type": "a",
                        "updateQueue": null,
                      },
                      "effectTag": 0,
                      "expirationTime": 0,
                      "firstEffect": null,
                      "index": 7,
                      "internalContextTag": 0,
                      "key": "7",
                      "lastEffect": null,
                      "memoizedProps": Object {
                        "children": <a
                          href="/demo"
>
                          Demo
</a>,
                      },
                      "memoizedState": null,
                      "nextEffect": null,
                      "pendingProps": null,
                      "ref": null,
                      "return": [Circular],
                      "sibling": null,
                      "stateNode": <li>
                        <a
                                                href="/demo"
                        >
                                                Demo
                        </a>
</li>,
                      "tag": 5,
                      "type": "li",
                      "updateQueue": null,
                    },
                    "stateNode": <li>
                      <a
                                            href="/loader"
                      >
                                            Loader
                      </a>
</li>,
                    "tag": 5,
                    "type": "li",
                    "updateQueue": null,
                  },
                  "stateNode": <li>
                    <a
                                        href="/tab-bars"
                    >
                                        Tab Bars
                    </a>
</li>,
                  "tag": 5,
                  "type": "li",
                  "updateQueue": null,
                },
                "stateNode": <li>
                  <a
                                    href="/list-views"
                  >
                                    List Views
                  </a>
</li>,
                "tag": 5,
                "type": "li",
                "updateQueue": null,
              },
              "stateNode": <li>
                <a
                                href="/cards"
                >
                                Cards
                </a>
</li>,
              "tag": 5,
              "type": "li",
              "updateQueue": null,
            },
            "stateNode": <li>
              <a
                            href="/buttons"
              >
                            Buttons
              </a>
</li>,
            "tag": 5,
            "type": "li",
            "updateQueue": null,
          },
          "stateNode": <li>
            <a
                        href="/bars"
            >
                        Bars
            </a>
</li>,
          "tag": 5,
          "type": "li",
          "updateQueue": null,
        },
        "stateNode": <li>
          <a
                    href="/alerts"
          >
                    Alerts
          </a>
</li>,
        "tag": 5,
        "type": "li",
        "updateQueue": null,
      },
      "effectTag": 0,
      "expirationTime": 0,
      "firstEffect": null,
      "index": 0,
      "internalContextTag": 0,
      "key": null,
      "lastEffect": null,
      "memoizedProps": Object {
        "children": Array [
          <li>
            <a
                        href="/alerts"
            >
                        Alerts
            </a>
</li>,
          <li>
            <a
                        href="/bars"
            >
                        Bars
            </a>
</li>,
          <li>
            <a
                        href="/buttons"
            >
                        Buttons
            </a>
</li>,
          <li>
            <a
                        href="/cards"
            >
                        Cards
            </a>
</li>,
          <li>
            <a
                        href="/list-views"
            >
                        List Views
            </a>
</li>,
          <li>
            <a
                        href="/tab-bars"
            >
                        Tab Bars
            </a>
</li>,
          <li>
            <a
                        href="/loader"
            >
                        Loader
            </a>
</li>,
          <li>
            <a
                        href="/demo"
            >
                        Demo
            </a>
</li>,
        ],
        "className": "sidebar",
      },
      "memoizedState": null,
      "nextEffect": null,
      "pendingProps": null,
      "ref": null,
      "return": [Circular],
      "sibling": null,
      "stateNode": <ul
        class="sidebar"
>
        <li>
                <a
                        href="/alerts"
                >
                        Alerts
                </a>
        </li>
        <li>
                <a
                        href="/bars"
                >
                        Bars
                </a>
        </li>
        <li>
                <a
                        href="/buttons"
                >
                        Buttons
                </a>
        </li>
        <li>
                <a
                        href="/cards"
                >
                        Cards
                </a>
        </li>
        <li>
                <a
                        href="/list-views"
                >
                        List Views
                </a>
        </li>
        <li>
                <a
                        href="/tab-bars"
                >
                        Tab Bars
                </a>
        </li>
        <li>
                <a
                        href="/loader"
                >
                        Loader
                </a>
        </li>
        <li>
                <a
                        href="/demo"
                >
                        Demo
                </a>
        </li>
</ul>,
      "tag": 5,
      "type": "ul",
      "updateQueue": null,
    },
    "effectTag": 1,
    "expirationTime": 0,
    "firstEffect": null,
    "index": 0,
    "internalContextTag": 0,
    "key": null,
    "lastEffect": null,
    "memoizedProps": Object {
      "links": Array [
        Object {
          "route": "/alerts",
          "text": "Alerts",
        },
        Object {
          "route": "/bars",
          "text": "Bars",
        },
        Object {
          "route": "/buttons",
          "text": "Buttons",
        },
        Object {
          "route": "/cards",
          "text": "Cards",
        },
        Object {
          "route": "/list-views",
          "text": "List Views",
        },
        Object {
          "route": "/tab-bars",
          "text": "Tab Bars",
        },
        Object {
          "route": "/loader",
          "text": "Loader",
        },
        Object {
          "route": "/demo",
          "text": "Demo",
        },
      ],
    },
    "memoizedState": Object {
      "active": false,
    },
    "nextEffect": null,
    "pendingProps": null,
    "ref": null,
    "return": FiberNode {
      "_debugID": 21,
      "_debugIsCurrentlyTiming": false,
      "_debugOwner": null,
      "_debugSource": null,
      "alternate": FiberNode {
        "_debugID": 21,
        "_debugIsCurrentlyTiming": false,
        "_debugOwner": null,
        "_debugSource": null,
        "alternate": [Circular],
        "child": null,
        "effectTag": 0,
        "expirationTime": 1,
        "firstEffect": null,
        "index": 0,
        "internalContextTag": 0,
        "key": null,
        "lastEffect": null,
        "memoizedProps": null,
        "memoizedState": null,
        "nextEffect": null,
        "pendingProps": null,
        "ref": null,
        "return": null,
        "sibling": null,
        "stateNode": Object {
          "containerInfo": <div>
            <ul
                        class="sidebar"
            >
                        <li>
                                    <a
                                                href="/alerts"
                                    >
                                                Alerts
                                    </a>
                        </li>
                        <li>
                                    <a
                                                href="/bars"
                                    >
                                                Bars
                                    </a>
                        </li>
                        <li>
                                    <a
                                                href="/buttons"
                                    >
                                                Buttons
                                    </a>
                        </li>
                        <li>
                                    <a
                                                href="/cards"
                                    >
                                                Cards
                                    </a>
                        </li>
                        <li>
                                    <a
                                                href="/list-views"
                                    >
                                                List Views
                                    </a>
                        </li>
                        <li>
                                    <a
                                                href="/tab-bars"
                                    >
                                                Tab Bars
                                    </a>
                        </li>
                        <li>
                                    <a
                                                href="/loader"
                                    >
                                                Loader
                                    </a>
                        </li>
                        <li>
                                    <a
                                                href="/demo"
                                    >
                                                Demo
                                    </a>
                        </li>
            </ul>
</div>,
          "context": Object {},
          "current": [Circular],
          "finishedWork": null,
          "hydrate": false,
          "isReadyForCommit": false,
          "nextScheduledRoot": null,
          "pendingChildren": null,
          "pendingContext": null,
          "remainingExpirationTime": 0,
        },
        "tag": 3,
        "type": null,
        "updateQueue": Object {
          "baseState": null,
          "callbackList": null,
          "expirationTime": 1,
          "first": Object {
            "callback": null,
            "expirationTime": 1,
            "isForced": false,
            "isReplace": false,
            "next": null,
            "nextCallback": null,
            "partialState": Object {
              "element": <Sidebar
                links={
                                Array [
                                                Object {
                                                  "route": "/alerts",
                                                  "text": "Alerts",
                                                },
                                                Object {
                                                  "route": "/bars",
                                                  "text": "Bars",
                                                },
                                                Object {
                                                  "route": "/buttons",
                                                  "text": "Buttons",
                                                },
                                                Object {
                                                  "route": "/cards",
                                                  "text": "Cards",
                                                },
                                                Object {
                                                  "route": "/list-views",
                                                  "text": "List Views",
                                                },
                                                Object {
                                                  "route": "/tab-bars",
                                                  "text": "Tab Bars",
                                                },
                                                Object {
                                                  "route": "/loader",
                                                  "text": "Loader",
                                                },
                                                Object {
                                                  "route": "/demo",
                                                  "text": "Demo",
                                                },
                                              ]
                }
/>,
            },
          },
          "hasForceUpdate": false,
          "isInitialized": false,
          "isProcessing": false,
          "last": Object {
            "callback": null,
            "expirationTime": 1,
            "isForced": false,
            "isReplace": false,
            "next": null,
            "nextCallback": null,
            "partialState": Object {
              "element": <Sidebar
                links={
                                Array [
                                                Object {
                                                  "route": "/alerts",
                                                  "text": "Alerts",
                                                },
                                                Object {
                                                  "route": "/bars",
                                                  "text": "Bars",
                                                },
                                                Object {
                                                  "route": "/buttons",
                                                  "text": "Buttons",
                                                },
                                                Object {
                                                  "route": "/cards",
                                                  "text": "Cards",
                                                },
                                                Object {
                                                  "route": "/list-views",
                                                  "text": "List Views",
                                                },
                                                Object {
                                                  "route": "/tab-bars",
                                                  "text": "Tab Bars",
                                                },
                                                Object {
                                                  "route": "/loader",
                                                  "text": "Loader",
                                                },
                                                Object {
                                                  "route": "/demo",
                                                  "text": "Demo",
                                                },
                                              ]
                }
/>,
            },
          },
        },
      },
      "child": [Circular],
      "effectTag": 0,
      "expirationTime": 0,
      "firstEffect": [Circular],
      "index": 0,
      "internalContextTag": 0,
      "key": null,
      "lastEffect": [Circular],
      "memoizedProps": null,
      "memoizedState": Object {
        "element": <Sidebar
          links={
                    Array [
                              Object {
                                "route": "/alerts",
                                "text": "Alerts",
                              },
                              Object {
                                "route": "/bars",
                                "text": "Bars",
                              },
                              Object {
                                "route": "/buttons",
                                "text": "Buttons",
                              },
                              Object {
                                "route": "/cards",
                                "text": "Cards",
                              },
                              Object {
                                "route": "/list-views",
                                "text": "List Views",
                              },
                              Object {
                                "route": "/tab-bars",
                                "text": "Tab Bars",
                              },
                              Object {
                                "route": "/loader",
                                "text": "Loader",
                              },
                              Object {
                                "route": "/demo",
                                "text": "Demo",
                              },
                            ]
          }
/>,
      },
      "nextEffect": null,
      "pendingProps": null,
      "ref": null,
      "return": null,
      "sibling": null,
      "stateNode": Object {
        "containerInfo": <div>
          <ul
                    class="sidebar"
          >
                    <li>
                              <a
                                        href="/alerts"
                              >
                                        Alerts
                              </a>
                    </li>
                    <li>
                              <a
                                        href="/bars"
                              >
                                        Bars
                              </a>
                    </li>
                    <li>
                              <a
                                        href="/buttons"
                              >
                                        Buttons
                              </a>
                    </li>
                    <li>
                              <a
                                        href="/cards"
                              >
                                        Cards
                              </a>
                    </li>
                    <li>
                              <a
                                        href="/list-views"
                              >
                                        List Views
                              </a>
                    </li>
                    <li>
                              <a
                                        href="/tab-bars"
                              >
                                        Tab Bars
                              </a>
                    </li>
                    <li>
                              <a
                                        href="/loader"
                              >
                                        Loader
                              </a>
                    </li>
                    <li>
                              <a
                                        href="/demo"
                              >
                                        Demo
                              </a>
                    </li>
          </ul>
</div>,
        "context": Object {},
        "current": [Circular],
        "finishedWork": null,
        "hydrate": false,
        "isReadyForCommit": false,
        "nextScheduledRoot": null,
        "pendingChildren": null,
        "pendingContext": null,
        "remainingExpirationTime": 0,
      },
      "tag": 3,
      "type": null,
      "updateQueue": null,
    },
    "sibling": null,
    "stateNode": [Circular],
    "tag": 2,
    "type": [Function],
    "updateQueue": null,
  },
  "_reactInternalInstance": Object {},
  "context": Object {},
  "props": Object {
    "links": Array [
      Object {
        "route": "/alerts",
        "text": "Alerts",
      },
      Object {
        "route": "/bars",
        "text": "Bars",
      },
      Object {
        "route": "/buttons",
        "text": "Buttons",
      },
      Object {
        "route": "/cards",
        "text": "Cards",
      },
      Object {
        "route": "/list-views",
        "text": "List Views",
      },
      Object {
        "route": "/tab-bars",
        "text": "Tab Bars",
      },
      Object {
        "route": "/loader",
        "text": "Loader",
      },
      Object {
        "route": "/demo",
        "text": "Demo",
      },
    ],
  },
  "refs": Object {},
  "state": Object {
    "active": false,
  },
  "updater": Object {
    "enqueueForceUpdate": [Function],
    "enqueueReplaceState": [Function],
    "enqueueSetState": [Function],
    "isMounted": [Function],
  },
}
`;
